### PR TITLE
Ftr:  Add an "api way" to set general configure

### PR DESCRIPTION
**Pull request by LaurenceLiZhixin**
**What this PR does**:
This pr add an "API way" to set general configure.
Usage can be found at my branch: 
https://github.com/LaurenceLiZhixin/dubbo-go-samples/blob/ftr/samples-add/config-api/go-client/cmd/client.go
or in test file

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: config/config_api.go [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+			Protocol: protocol,
+		}
+	}
+}
+
+///////////////////////////////////// registry config api
+type RegistryConfigOpt func(config *RegistryConfig) *RegistryConfig
+
+func NewRegistryConfig(opts ...RegistryConfigOpt) *RegistryConfig {
+	newRegistryConfig := NewDefaultRegistryConfig("none")
```
Comment (gaoxinge) [inline config/config_api.go:64]: Use empty string "" instead of "none".

File: config/config_api.go [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+		return config
+	}
+}
+func WithReferenceCluster(cluster string) ReferenceConfigOpt {
+	return func(config *ReferenceConfig) *ReferenceConfig {
+		config.Cluster = cluster
+		return config
+	}
+}
+func WithReferenceMethod(methodName, retries, lb string) ReferenceConfigOpt {
```
Comment (gaoxinge) [inline config/config_api.go:262]: Add blank between functions.

Comment (LaurenceLiZhixin) [reply config/config_api.go:267]: fixed!

File: config/config_api.go (hunk 1 above)
Comment (LaurenceLiZhixin) [reply config/config_api.go]: fixed!

File: config/config_api.go [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+		config.Methods = append(config.Methods, &MethodConfig{
+			Name:        name,
+			Retries:     retries,
+			LoadBalance: lb,
+		})
+		return config
+	}
+}
+
+///////////////////////////////////////// Application config api
```
Comment (AlexStocks) [inline config/config_api.go:416]: // enough

**Review by AlexStocks — APPROVED** [review]

**Review by gaoxinge — APPROVED** [review]

**Review by zouyx — COMMENTED** [review]
Base on security, I am not agree with modify configuration by web API .

**Comment by AlexStocks** [comment]
pls change the target branch to 1.5

**Comment by gaoxinge** [comment]
This pull request may correspond to the [dubbo java's api configuration](https://dubbo.apache.org/docs/v2.7/user/configuration/api/). I think this is a nice configuration feature for people who start to use dubbo-go.  @AlexStocks @zouyx

**Comment by zouyx** [comment]
> This pull request may correspond to the [dubbo java's api configuration](https://dubbo.apache.org/docs/v2.7/user/configuration/api/). I think this is a nice configuration feature for people who start to use dubbo-go. @AlexStocks @zouyx

I will agree,if not web API .

**Review by zouyx — APPROVED** [review]

**Comment by AlexStocks** [comment]
very good pr. however, u should add comment for every func.

**Pull request merged**